README
***************************************
Cire: An Open-Source Diary Program in C

Copyright (C) 2004, Andrew Morritt, <andymorritt@rogers.com>, et al.

This software is licensed as described in the file COPYING, which
you should have received as part of this distribution. The terms
are also available at http://cire.sourceforge.net/license.php

You may opt to use, copy, modify, merge, publish, distribute and/or sell
copies of the Software, and permit persons to whom the Software is
furnished to do so, under the terms of the COPYING file.

This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
KIND, either express or implied.

File is Managed by Andrew Morritt (andrewm1). Get Permission Before Commiting.

$Id: README,v 1.15 2004/11/24 02:54:44 andrewm1 Exp $
***************************************

Contents:

	1: Compilation & Installation 

	2: Use:

	3: The Cire Project:

	4: Tips:

	5: Known Issues:
	
	6: Contributors & Contributing:

	7: ChangeLog
	
	8: Legalese

***************************

1: Compilation & Installation:
	See the file INSTALL

2: Use: 

	Start Cire with the UI of your choice. Use Cire. It's that simple!
	
	You can use:
	
	'cire -g <diary filename>' to Start GTK Interface,
	'cire -n <diary filename>' to Start Ncurses Interface,
	'cire -t <diary filename>' to Start Text Interface,

3: The Cire Project:

	The Cire Project can be contacted at http://cire.sf.net
	The Dev Home is at http://sf.net/projects/cire/

4: Tips:
	* None Right Now

5: Known Issues:
	See the file TODO
	
6: Contributors & Contributing:
	A List of contributors is avaliable in the file CREDITS.

	Information on how to Contribute is Avaliable in the 
	file HACKING.

7: ChangeLog:
	See the file CHANGES

8: Legalese:
	The License Governing The Distribution and Use of 
	Cire is available in the file COPYING

Thank You For Using Cire!


CHANGES
***************************************
Cire: An Open-Source Diary Program in C

Copyright (C) 2004, Andrew Morritt, <andymorritt@rogers.com>, et al.

This software is licensed as described in the file COPYING, which
you should have received as part of this distribution. The terms
are also available at http://cire.sourceforge.net/license.php

You may opt to use, copy, modify, merge, publish, distribute and/or sell
copies of the Software, and permit persons to whom the Software is
furnished to do so, under the terms of the COPYING file.

This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
KIND, either express or implied.

File is Managed by Andrew Morritt (andrewm1). Get Permission Before Commiting.

$Id: CHANGES,v 1.12 2004/11/24 02:54:43 andrewm1 Exp $
***************************************

Cire Version 0.14.0:

	2004-10-16 17:07  thestarz

		* Makefile.am: Andrew put a list of all source files in
		  /Makefile.am as well as /src/Makefile.am. I've fixed this.

	2004-10-14 19:40  andrewm1

		* README: Removed Periods from End of URL's

	2004-10-14 15:39  andrewm1

		* CREDITS, HACKING, README, TODO, configure.ac: - Added Tom Davis
		  To CREDITS

		  - Updated HACKING for new Autotools Stuff

		  - README *really* needed updating

		  - Added some Template stuff for a --with-debugging in
		  configure.ac,     see TODO for more intop. If it gets in, I'll
		  add the option to the list	     in INSTALL.

		  - To Conform with Automake, I was considering asking the SF
		  People to   rename CHANGES to ChangeLog and CREDITS to AUTHORS,
		  what	      do other people think?

	2004-10-14 11:19  andrewm1

		* CHANGES, COPYING, CREDITS, HACKING, INSTALL, Makefile.am, README,
		  TODO, autogen.sh, configure.ac, src/Makefile.am, src/diary.c,
		  src/diary.h, src/entry.c, src/entry.h, src/gtk_ui_callbacks.c,
		  src/gtk_ui_callbacks.h, src/gtk_ui_main.c, src/gtk_ui_main.h,
		  src/gtk_ui_other.c, src/gtk_ui_other.h, src/main.c, src/main.h,
		  src/misc.c, src/misc.h, src/ncurses_ui.c, src/ncurses_ui.h,
		  src/prefs.c, src/prefs.h, src/text_ui.c, src/text_ui.h: New
		  Copyright Message, Cause that's what I do Best! :-)

	2004-10-13 17:54  andrewm1

		* CHANGES, COPYING, CREDITS, HACKING, INSTALL, Makefile.am, README,
		  TODO, autogen.sh, build.pl, configure.ac, src/Makefile.am,
		  src/diary.c, src/diary.h, src/entry.c, src/entry.h,
		  src/gtk_ui_callbacks.c, src/gtk_ui_callbacks.h,
		  src/gtk_ui_main.c, src/gtk_ui_main.h, src/gtk_ui_other.c,
		  src/gtk_ui_other.h, src/main.c, src/main.h, src/misc.c,
		  src/misc.h, src/ncurses_ui.c, src/ncurses_ui.h, src/prefs.c,
		  src/prefs.h, src/text_ui.c, src/text_ui.h: Big, BIG CHANGES!!!

		  First of All, I added in the Autotools system by John. This, in
		  essence, has completely reorganized the code and made the
		  intallation much nicer.

		  Also, I added Id Tags to all files... Hopefully, it shall all
		  work :-)

		  I think we have enough changes in here for the next version...
		  Perhaps... Ah well, we'll discuss it.

	2004-04-21 20:05  thestarz

		* src/: gtk_ui_callbacks.c, gtk_ui_callbacks.h, gtk_ui_main.c,
		  gtk_ui_main.h, gtk_ui_other.c, gtk_ui_other.h: Applied Tom's
		  patch adding search support to the gtk interface. Sorry it took
		  so long, I kind forgot about it...

	2004-04-15 21:06  andrewm1

		* CREDITS:

		  New Credits Format!

	2004-04-14 12:59  thestarz

		* TODO, src/misc.c, src/misc.h, src/ncurses_ui.c: Search is now not
		  case sensetive. For some reason though, searching for a string of
		  less than four characters doesn't seem to work. If anybody can
		  figure out why, let me know.

			  John

	2004-04-13 12:55  thestarz

		* TODO, src/diary.c, src/diary.h, src/ncurses_ui.c,
		  src/ncurses_ui.h: Applied Ian's patch for exporting to LaTeX.
		  Good code Ian, however in some places you had spaces instead of
		  tabs for indentation. Try to avoid that ;)

			  John

	2004-04-12 19:37  thestarz

		* TODO, src/ncurses_ui.c: The search prompt in the text viewer no
		  longer displays the list of entries instead of the text being
		  view. I changed nci_get_user_text to not update the main part of
		  the screen any more. This, however, led to the problem of the
		  screen being messed up if the term is resized while a text prompt
		  is open. To solve this I made it clear the main part of the
		  screen if, and only if, the term is resized.

		  I also made some changed to TODO. Take a look.

			  John

	2004-04-11 20:00  thestarz

		* TODO, src/ncurses_ui.c: Added search support to the ncurses text
		  (entry) viewer.

			  John

	2004-04-11 12:53  thestarz

		* src/: diary.c, entry.c, main.c, main.h, misc.c, ncurses_ui.c,
		  ncurses_ui.h: Changed the method used by the core code to report
		  errors.

		  Previously when something went wrong, an error message was
		  printed to stderr. However with the gtk interface this would
		  never be seen and with the ncurses interface it messed up the
		  screen.

		  Now the error message is placed in the global variable char*
		  errmsg. Then the interface (ncurses / gtk) can display it in the
		  appropriate way (in the footer area or in a pop up dialog).

		  I also created a new function for the ncurses interface:
		  nci_press_any_key.  It takes a message which it prints to the
		  footer area followed by "Press any key to continue." and waits
		  for a key press before returning. The ncurses interface now uses
		  this with errmsg to tell the user what went if the core code
		  returns an error code.

			  John

	2004-04-10 17:47  thestarz

		* src/ncurses_ui.c: Fixed a bug in the code for getting text from
		  the user using the footer area. Previously it would segfault if
		  you enlarged your terminal after getting to that prompt and then
		  typing a lot of text into it.

			  John

	2004-04-10 15:40  thestarz

		* TODO, src/ncurses_ui.c: Page up and page down now scroll the list
		  of entries up or down one page.

			  John


src/main.h
/*
 * ***************************************
 * Cire: An Open-Source Diary Program in C
 *
 * Copyright (C) 2004, Andrew Morritt, <andymorritt@rogers.com>, et al.
 *
 * This software is licensed as described in the file COPYING, which
 * you should have received as part of this distribution. The terms
 * are also available at http://cire.sourceforge.net/license.php
 *
 * You may opt to use, copy, modify, merge, publish, distribute and/or sell
 * copies of the Software, and permit persons to whom the Software is
 * furnished to do so, under the terms of the COPYING file.
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
 * KIND, either express or implied.
 *
 * File is a General Cire File. Anyone may commit changes.
 *
 * $Id: main.h,v 1.22 2004/11/24 02:54:44 andrewm1 Exp $
 *  ***************************************
*/

#ifndef __MAIN_H
#define __MAIN_H

#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
#include <time.h>
#include <errno.h>

#ifdef BUILD_NCURSES_UI
	#include <curses.h>
#endif

#ifdef BUILD_GTK_UI
	#include <gtk/gtk.h>
#endif

#include "misc.h"
#include "entry.h"
#include "diary.h"
#include "prefs.h"
#include "ncurses_ui.h"
#include "text_ui.h"
#include "gtk_ui_main.h"
#include "gtk_ui_callbacks.h"
#include "gtk_ui_other.h"

#define CIRE_VERSION "0.14.0"

/* this is an 'about' message, it's shared by the ncurses and gtk interfaces */
#define CIRE_ABOUT_MESSAGE cire_about_message
const char* cire_about_message;

char* errmsg;
/* When an error is encountered in the core code, it puts an error message
 * here for the current user interface to display to the user however it wants.
 */

#endif
